xwiki-platform-core/xwiki-platform-security/xwiki-platform-security-authentication/xwiki-platform-security-authentication-default/src/test/java/org/xwiki/security/authentication/internal/DefaultResetPasswordManagerTest.java

```diff
@@ -464,150 +466,199 @@ void resetPasswordNotDocumentReferenceUser() throws Exception
-        assertFalse(this.resetPasswordManager.isPasswordCompliantWithRegistrationRules("_/3434"));
+        when(this.registrationConfiguration.getPasswordMinimumLength()).thenReturn(minimumLength);
+        when(this.registrationConfiguration.getPasswordRules()).thenReturn(rules);
+        assertEquals(expectation, this.resetPasswordManager.isPasswordCompliantWithRegistrationRules(password));
+    }
+
+    private static Stream<Arguments> provideArgumentsForIsPasswordCompliantWithRegistrationRules()
+    {
+        Set<RegistrationConfiguration.PasswordRules> noRules = Set.of();
+        Set<RegistrationConfiguration.PasswordRules> lowerMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_LOWER_CASE_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> upperMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> symbolMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_SYMBOL_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> numberMandatory =
+            Set.of(RegistrationConfiguration.PasswordRules.ONE_NUMBER_CHARACTER);
+        Set<RegistrationConfiguration.PasswordRules> upperAndNumberMandatory =
+            Set.of(
+                RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_NUMBER_CHARACTER
+            );
+        Set<RegistrationConfiguration.PasswordRules> lowerAndUpperMandatory =
+            Set.of(
+                RegistrationConfiguration.PasswordRules.ONE_LOWER_CASE_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER
+            );
+        Set<RegistrationConfiguration.PasswordRules> allRules =
+            Set.of(
+                RegistrationConfiguration.PasswordRules.ONE_LOWER_CASE_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_UPPER_CASE_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_NUMBER_CHARACTER,
+                RegistrationConfiguration.PasswordRules.ONE_SYMBOL_CHARACTER
+            );
+
+        String onlyLower3Chars = "foo";
+        String onlyLower6Chars = "fooooo";
+        String lowerUpper11Chars = "foooOOOOOoo";
+        String lowerNumber6Chars = "foo000";
+        String lowerUpperNumber11Chars = "fooOOOOo000";
+        String lowerNumberSymbolSpace9Chars = "foo_/ 000";
+        String lowerSymbolNoSpace5Chars = "foo_/";
+        String lowerUpperSymbolNoSpace9Chars = "fOOOOoo_/";
+        String lowerUpperSymbolNumberSpace12Chars = "foOOOo_/ 000";
+        String lowerUpperSymbolNumberSpaceAccents14Chars = "foOOOo_/ éé000";
+        String symbolSpace3Chars = " _/";
+        String symbolNumberNoSpace6Chars = "_/3434";
+        String onlyUpper3Chars = "FOO";
+        String onlyUpper6Chars = "FOOOOO";
+
+        return Stream.of(
```

Comment:
That's an impressive list of parameters :)

Reply:
Yeah, I might have done it otherwise to have a  shorter list but I find it more readable like this.